CMakeLists.txt

```diff
@@ -16,40 +16,45 @@
 
 # Note: on OS X you should install XCode and the associated command-line tools
 
-PROJECT("paho" C)
-CMAKE_MINIMUM_REQUIRED(VERSION 3.0)
+project("paho" C)
+cmake_minimum_required(VERSION 3.0)
+
+if (NOT CMAKE_TOOLCHAIN_FILE)
+    message(FATAL_ERROR
+        "Paho has to be built using a toolchain file. \
+These can be found in ${CMAKE_CURRENT_LIST_DIR}/cmake and can be \
+specified using the CMAKE_TOOLCHAIN_FILE variable when invoking cmake."
+    )
+endif ()
 
 ## build settings
-SET(PAHO_VERSION_MAJOR 1)
-SET(PAHO_VERSION_MINOR 0)
-SET(PAHO_VERSION_PATCH 3)
-SET(CLIENT_VERSION ${PAHO_VERSION_MAJOR}.${PAHO_VERSION_MINOR}.${PAHO_VERSION_PATCH})
+file(READ version.major PAHO_VERSION_MAJOR)
+file(READ version.minor PAHO_VERSION_MINOR)
+file(READ version.patch PAHO_VERSION_PATCH)
+set(CLIENT_VERSION ${PAHO_VERSION_MAJOR}.${PAHO_VERSION_MINOR}.${PAHO_VERSION_PATCH})
 
-EXECUTE_PROCESS(COMMAND date -u OUTPUT_VARIABLE BUILD_TIMESTAMP)
-STRING(STRIP ${BUILD_TIMESTAMP} BUILD_TIMESTAMP)
+string(TIMESTAMP BUILD_TIMESTAMP UTC)
 
 ## build options
-SET(PAHO_WITH_SSL FALSE CACHE BOOL "Flag that defines whether to build ssl-enabled binaries too. ")
-SET(PAHO_BUILD_DOCUMENTATION FALSE CACHE BOOL "Create and install the HTML based API documentation (requires Doxygen)")
-SET(PAHO_BUILD_SAMPLES FALSE CACHE BOOL "Build sample programs")
+set(PAHO_WITH_SSL FALSE CACHE BOOL "Flag that defines whether to build ssl-enabled binaries too. ")
+set(PAHO_BUILD_DOCUMENTATION FALSE CACHE BOOL "Create and install the HTML based API documentation (requires Doxygen)")
+set(PAHO_BUILD_SAMPLES FALSE CACHE BOOL "Build sample programs")
 
```

Comment:
You could use the 'option' command to have those build options readily available and visible e.g. in CMake gui on first build. Those can still be overwritten from an outside build system like buildroot or ptxdist.